[vulkan] Add nonVarTypeModeGuard to vulkan tests and speed_benchmark_torch

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52535 [vulkan] Add nonVarTypeModeGuard to vulkan tests and speed_benchmark_torch**

Differential Revision: [D26580994](https://our.internmc.facebook.com/intern/diff/D26580994)

Some ops with Math kernel registrations are now failing with

```
C++ exception with description "0 INTERNAL ASSERT FAILED at "/Users/ssjia/pytorch/aten/src/ATen/core/boxing/KernelFunction.cpp":30, please report a bug to PyTorch. aten::adaptive_avg_pool2d has kernels registered to both Math and a backend mapped to AutogradOther. This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). If it's intended to override Math kernel behavior, please open an issue to request a dedicated Autograd dispatch key for the backend.
If you only want to run inference instead of training, add `at::AutoNonVariableTypeMode guard(true);` before model.forward(). Note this guard is only available in C++ but not Python at present.
Canonical state
~~~~~~~~~~~
name: aten::adaptive_avg_pool2d
schema: aten::adaptive_avg_pool2d(Tensor self, int[2] output_size) -> (Tensor)
debug: registered at /Users/ssjia/pytorch/build/aten/src/ATen/RegisterSchema.cpp:20
alias analysis kind: FROM_SCHEMA
Metal: registered at /Users/ssjia/pytorch/aten/src/ATen/native/metal/MetalAten.mm:263 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
Tracer: registered at /Users/ssjia/pytorch/torch/csrc/autograd/generated/TraceType_1.cpp:11324 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
Math[alias]: registered at /Users/ssjia/pytorch/build/aten/src/ATen/RegisterMath.cpp:5686 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
```

Add the variable guard to prevent this error.

Commit:
[vulkan] Add nonVarTypeModeGuard to vulkan tests and speed_benchmark_torch

[ghstack-poisoned]

AshkanAliabadi commented:
@dreiss @ljk53 @smessmer Can one of you guys please take a look?  I am not familiar with these stuff and would like to know whether we are doing the right thing here.  Thanks!

ailzhang reviewed (approved):
Talked to Stephen on workchat that these tests are not enabled in CI, so it's likely failing for a while.
Once we have inference_mode guard we should update these callsites.

Commit:
Update on "[vulkan] Add nonVarTypeModeGuard to vulkan tests and speed_benchmark_torch"


Differential Revision: [D26580994](https://our.internmc.facebook.com/intern/diff/D26580994)

Some ops with Math kernel registrations are now failing with

```
C++ exception with description "0 INTERNAL ASSERT FAILED at "/Users/ssjia/pytorch/aten/src/ATen/core/boxing/KernelFunction.cpp":30, please report a bug to PyTorch. aten::adaptive_avg_pool2d has kernels registered to both Math and a backend mapped to AutogradOther. This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). If it's intended to override Math kernel behavior, please open an issue to request a dedicated Autograd dispatch key for the backend.
If you only want to run inference instead of training, add `at::AutoNonVariableTypeMode guard(true);` before model.forward(). Note this guard is only available in C++ but not Python at present.
Canonical state
~~~~~~~~~~~
name: aten::adaptive_avg_pool2d
schema: aten::adaptive_avg_pool2d(Tensor self, int[2] output_size) -> (Tensor)
debug: registered at /Users/ssjia/pytorch/build/aten/src/ATen/RegisterSchema.cpp:20
alias analysis kind: FROM_SCHEMA
Metal: registered at /Users/ssjia/pytorch/aten/src/ATen/native/metal/MetalAten.mm:263 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
Tracer: registered at /Users/ssjia/pytorch/torch/csrc/autograd/generated/TraceType_1.cpp:11324 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
Math[alias]: registered at /Users/ssjia/pytorch/build/aten/src/ATen/RegisterMath.cpp:5686 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
```

Add the variable guard to prevent this error.

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add nonVarTypeModeGuard to vulkan tests and speed_benchmark_torch"


Differential Revision: [D26580994](https://our.internmc.facebook.com/intern/diff/D26580994)

Some ops with Math kernel registrations are now failing with

```
C++ exception with description "0 INTERNAL ASSERT FAILED at "/Users/ssjia/pytorch/aten/src/ATen/core/boxing/KernelFunction.cpp":30, please report a bug to PyTorch. aten::adaptive_avg_pool2d has kernels registered to both Math and a backend mapped to AutogradOther. This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). If it's intended to override Math kernel behavior, please open an issue to request a dedicated Autograd dispatch key for the backend.
If you only want to run inference instead of training, add `at::AutoNonVariableTypeMode guard(true);` before model.forward(). Note this guard is only available in C++ but not Python at present.
Canonical state
~~~~~~~~~~~
name: aten::adaptive_avg_pool2d
schema: aten::adaptive_avg_pool2d(Tensor self, int[2] output_size) -> (Tensor)
debug: registered at /Users/ssjia/pytorch/build/aten/src/ATen/RegisterSchema.cpp:20
alias analysis kind: FROM_SCHEMA
Metal: registered at /Users/ssjia/pytorch/aten/src/ATen/native/metal/MetalAten.mm:263 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
Tracer: registered at /Users/ssjia/pytorch/torch/csrc/autograd/generated/TraceType_1.cpp:11324 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
Math[alias]: registered at /Users/ssjia/pytorch/build/aten/src/ATen/RegisterMath.cpp:5686 :: (Tensor _0, int[] _1) -> (Tensor _0) [ boxed unboxed ]
```

Add the variable guard to prevent this error.

[ghstack-poisoned]
